src/librustdoc/html/highlight.rs

```diff
@@ -41,20 +47,29 @@ crate fn render_with_highlighting(
     }
 
     write_header(out, class);
-    write_code(out, &src, edition);
+    write_code(out, &src, edition, file_span_lo, context, root_path);
     write_footer(out, playground_button);
 }
 
 fn write_header(out: &mut Buffer, class: Option<&str>) {
     write!(out, "<div class=\"example-wrap\"><pre class=\"rust {}\">\n", class.unwrap_or_default());
 }
 
-fn write_code(out: &mut Buffer, src: &str, edition: Edition) {
+fn write_code(
```

Comment:
Document file_span_lo please, and the behavior when context is None (and elsewhere in the PR)